platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaExecWithLongCommandLineIntegrationTest.groovy

```diff
@@ -49,7 +49,9 @@ class JavaExecWithLongCommandLineIntegrationTest extends AbstractIntegrationSpec
                 dependsOn sourceSets.main.runtimeClasspath
                 doLast {
                     project.javaexec {
-                        executable = run.executable
+                        if (run.executable.isPresent()) {
```

Comment:
💭 This reverts behaviour to the same as is in `master`. If we set executable to empty provider, then execution will fail since executable is not set